x-pack/plugin/stateless/src/main/java/org/elasticsearch/xpack/stateless/StatelessPlugin.java

```diff
@@ -135,6 +150,8 @@ public Settings additionalSettings() {
             return Settings.builder()
                 .put(super.additionalSettings())
                 .put(CLUSTER_ROUTING_ALLOCATION_DISK_THRESHOLD_ENABLED_SETTING.getKey(), false)
+                .put(DATA_STREAMS_LIFECYCLE_ONLY_MODE.getKey(), true)
```

Comment:
Would you like to extend `testSettingsWithDefaultDiskThresholdEnabled` to check these settings' values as well?